src/primitives.cc

```diff
@@ -96,6 +96,7 @@ class PrimitiveNode : public LeafNode
 			return "unknown";
 		}
 	}
+	const Context *ctx;
```

Comment:
As mentioned earlier: Are you certain that this object will survive the PrimitiveNode instance?